internal/discovery/cache.go

```diff
@@ -12,16 +12,16 @@ import (
 	"k8s.io/kube-openapi/pkg/util/proto"
 )
 
-// discoveryCache is useful to prevent excessive QPS to the discovery APIs while
+// Cache is useful to prevent excessive QPS to the discovery APIs while
 // still allowing dynamic refresh of the openapi spec on cache misses.
-type discoveryCache struct {
+type Cache struct {
```

Comment:
The discovery cache has very limited surface area with the controller, so it's a good candidate to have its own package.